logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/client/RegisterClientUseCase.kt

```diff
@@ -166,18 +176,24 @@ class RegisterClientUseCaseImpl @OptIn(DelicateKaliumApi::class) constructor(
             kaliumException.isBadRequest() ->
                 RegisterClientResult.Failure.InvalidCredentials.InvalidPassword
 
-            authCodeFailure != null -> when (authCodeFailure) {
-                AuthenticationCodeFailure.MISSING_AUTHENTICATION_CODE ->
-                    RegisterClientResult.Failure.InvalidCredentials.Missing2FA
-
-                AuthenticationCodeFailure.INVALID_OR_EXPIRED_AUTHENTICATION_CODE ->
-                    RegisterClientResult.Failure.InvalidCredentials.Invalid2FA
-            }
+            authCodeFailure != null -> handleAuthCodeFailure(authCodeFailure)
 
             else -> RegisterClientResult.Failure.Generic(failure)
         }
     } else RegisterClientResult.Failure.Generic(failure)
 
+    private suspend fun handleAuthCodeFailure(authCodeFailure: AuthenticationCodeFailure) = when (authCodeFailure) {
+        AuthenticationCodeFailure.MISSING_AUTHENTICATION_CODE ->
+            RegisterClientResult.Failure.InvalidCredentials.Missing2FA
+
+        AuthenticationCodeFailure.INVALID_OR_EXPIRED_AUTHENTICATION_CODE -> {
+            userRepository.getSelfUser()?.email?.let {
+                secondFactorVerificationRepository.clearStoredVerificationCode(it)
+            }
+            RegisterClientResult.Failure.InvalidCredentials.Invalid2FA
+        }
+    }
+
```

Comment:
same here, it has a return type so I am expecting it to simply returns something and do not have a side effect I think it would make more sense to split it in two steps and leave the handleAuthCodeFailure() side-effect free